src/store.ts

```diff
@@ -148,8 +143,49 @@ const useAppStore = create<AppState>()(
         }
         set(() => ({ data }));
       },
+      generateShareableLink: () => {
+        const state = get();
+        const compressedData = compress({
+          templateMarkdown: state.templateMarkdown,
+          modelCto: state.modelCto,
+          data: state.data,
+        });
+        return `${window.location.origin}?data=${compressedData}`;
+      },
+      loadFromLink: async (compressedData: string) => {
+        try {
+          const { templateMarkdown, modelCto, data } =
```

Comment:
@sanketshevkar @Vinyl-Davyl Is there a good reason not to use the template archive JSON here instead?

If so, do you expect to support links generated with this format in addition to template archives in the future?

Reply:
I have suggested that we look into versioning the URLs. We start with v1 having a simple object compressed and added to URL. v2 we can introduce template archive JSON.
I didn't want go down the template-archive route right now, given the time left for the program I concluded it could take some more time for David to onboard with template-archive. If time permits we could develop a v2 version of URL with template-archive object.

Reply:
Thank you, @sanketshevkar  and @mttrbrts.

I agree with Sanket's suggestion. Given the current time constraints, starting with a simpler object compressed and added to the URL for version 1 allows us to implement a functional solution more swiftly. Perhaps might enable us to meet our immediate goals while ensuring that we have a solid foundation to build upon.

Introducing template archive JSON in a future version (v2) would be an excellent idea. Once we have more time, I am confident we can develop and implement the template-archive format effectively.

For now, just trying to focus on the initial implementation to help us deliver a usable product on schedule.